include/lbann/weights/weights.hpp

```diff
@@ -77,6 +77,8 @@ class weights {
   /** Get weights name. */
   std::string get_name() const { return m_name; }
 
+  lbann_comm* get_comm() const { return m_comm; }
```

Comment:
We should return a reference. If `m_comm` is a null pointer, we should throw an exception.

Reply:
Done.